indexdata.go

```diff
@@ -236,17 +236,18 @@ func (d *indexData) memoryUse() int {
 	for _, a := range [][]uint32{
 		d.newlinesIndex, d.docSectionsIndex,
 		d.boundaries, d.fileNameIndex,
-		d.runeOffsets, d.fileNameRuneOffsets,
 		d.fileEndRunes, d.fileNameEndRunes,
 		d.fileEndSymbol, d.symbols.symKindIndex,
 	} {
 		sz += 4 * len(a)
 	}
-	sz += 8 * len(d.runeDocSections)
+	sz += d.runeOffsets.sizeBytes()
+	sz += d.fileNameRuneOffsets.sizeBytes()
+	sz += len(d.runeDocSections)
 	sz += 8 * len(d.fileBranchMasks)
-	sz += 12 * len(d.ngrams)
+	sz += d.ngrams.SizeBytes()
 	for _, v := range d.fileNameNgrams {
-		sz += 4*len(v) + 4
+		sz += len(v) + 4
```

Comment:
I don't think this is correct. `[]byte` is from the mmap region, so does not directly contribute to go's heap. I think you can just do `sz += sizeof([]byte) * len(d.fileNameNgrams)`. You may want to instead make `fileNameNgrams` point to `simpleSection` which I think is smaller than a `[]byte`.